src/Microsoft.AspNetCore.HttpsPolicy/HttpsRedirectionBuilderExtensions.cs

```diff
@@ -31,10 +32,25 @@ public static IApplicationBuilder UseHttpsRedirection(this IApplicationBuilder a
 
             var options = app.ApplicationServices.GetRequiredService<IOptions<HttpsRedirectionOptions>>().Value;
 
+            // The tls port set in options will have priority over the one in configuration.
+            var tlsPort = options.SSLPort;
+            if (tlsPort == null)
+            {
+                // Only read configuration if there is no tlsPort
+                var config = app.ApplicationServices.GetRequiredService<IConfiguration>();
+                var configTlsPort = config["SSL_PORT"];
```

Comment:
@tratcher Should this be TLS_PORT or TLSPORT or HTTPSPORT or HTTPS_PORT

Reply:
I'm okay with changing it to HTTPS_PORT, but I agree with @Tratcher for SSL > TLS.